jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -519,3 +526,15 @@ def _live_update(self, event={}):
         for mark in self.marks.values():
             mark.update_xy(sp.spectral_axis.value, sp.flux.value)
             mark.visible = True
+
+    def translate_units(self, collapsed_spec):
+        # MJy/sr -> Jy/pix
+        if collapsed_spec._unit == u.MJy / u.sr:
+            collapsed_spec._data *= collapsed_spec.meta['_pixel_scale_factor']
+            collapsed_spec._unit = u.Jy / u.pix
+        # Jy/pix -> MJy/sr
+        elif collapsed_spec._unit == u.Jy / u.pix:
+            collapsed_spec._data /= collapsed_spec.meta['_pixel_scale_factor']
+            collapsed_spec._unit = u.MJy / u.sr
```

Comment:
I still think we want this to be generalized to any surface brightness -> flux units, basically just adding or removing the steradians but keeping the same base units otherwise (which probably means we don't need the 10^6 in the factor).